Maestro/src/Microsoft.DotNet.Maestro.WebApi/Controllers/CommitPushedController.cs

```diff
@@ -82,5 +96,82 @@ public class CommitPushedController : ApiController
+            {
+                Trace.TraceWarning($"Payload signature: {payloadSignature}");
+                Trace.TraceWarning($"Computed hash: {computedHash}");
+
+                throw new Exception("The payload signature specified by the X-Hub-Signature header does not match " +
+                    "the hash computed from the request.  This could indicate that the repo's Maestro webhook has " +
+                    "an incorrect Secret, the request came from an unauthorized source, or the request was corrupted.");
+            }
+        }
+
+        private string GetPayloadSignature()
+        {
+            IEnumerable<string> headerValues;
+            string requestXHubSignature = string.Empty;
+            if (Request.Headers.TryGetValues("X-Hub-Signature", out headerValues))
+            {
+                requestXHubSignature = headerValues.FirstOrDefault();
+            }
+
+            if (string.IsNullOrWhiteSpace(requestXHubSignature) || !requestXHubSignature.StartsWith("sha1="))
+            {
+                throw new Exception("The HTTP request does not contain a valid X-Hub-Signature header.  Verify that " +
+                    "the repo's Maestro webhook is configured with a Secret.");
+            }
+
+            return requestXHubSignature.Substring(5).ToUpper();
+        }
+
+        private string ComputeHashHexdigest(byte[] requestContent)
+        {
+            byte[] hmacKey = Encoding.UTF8.GetBytes(Config.Instance.WebhookSecretToken);
+
+            using (HMACSHA1 hmac = new HMACSHA1(hmacKey))
+            {
+                byte[] computedHash = hmac.ComputeHash(requestContent);
+                return BitConverter.ToString(computedHash).Replace("-", string.Empty);
+            }
+        }
+
+        // Test two strings for equality using a constant-time algorithm such that the execution time varies
+        // only on the length of the data, not the contents thereof, to prevent timing attacks. This method
+        // assumes that the strings are indexable.
+        [MethodImpl(MethodImplOptions.NoInlining | MethodImplOptions.NoOptimization)]
+        private static bool CryptographicEquals(string a, string b)
+        {
+            int result = 0;
+
+            // Short cut if the lengths are not identical
+            if (a.Length != b.Length)
+            {
+                return false;
+            }
+
+            unchecked
+            {
+                // Normally this caching doesn't matter, but with the optimizer off, this nets a non-trivial
+                // speedup.
+                int aLength = a.Length;
+
+                for (int i = 0; i < aLength; i++)
+                    // We use subtraction here instead of XOR because the XOR algorithm gets ever so
+                    // slightly faster as more and more differences pile up.
+                    // This cannot overflow more than once (and back to 0) because bytes are 1 byte
```

Comment:
This comment is no longer correct because you are not using bytes, you are using chars.